androidApp/src/main/java/com/example/tasky/android/MainNavHost.kt

```diff
@@ -1,18 +1,47 @@
 package com.example.tasky.android
 
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
 import androidx.compose.ui.Modifier
 import androidx.navigation.NavHostController
 import androidx.navigation.compose.NavHost
+import com.example.tasky.android.agenda.AgendaGraph
+import com.example.tasky.android.agenda.agendaGraph
+import com.example.tasky.android.agenda.navigateToAgendaGraph
 import com.example.tasky.android.login.LoginGraph
 import com.example.tasky.android.login.loginGraph
+import com.example.tasky.android.login.navigateToLoginGraph
+import com.example.tasky.manager.loginManager
 
 @Composable
 fun MainNavHost(
     navHostController: NavHostController,
     modifier: Modifier = Modifier,
 ) {
-    NavHost(navController = navHostController, startDestination = LoginGraph, modifier = modifier) {
-        loginGraph(navHostController)
+    LaunchedEffect(true) {
+        loginManager.isLoginFlow.collect { isLogin ->
+            if (isLogin) {
+                navHostController.navigateToAgendaGraph()
+            } else {
+                navHostController.navigateToLoginGraph()
+            }
+        }
+    }
+
+    val isLogin = loginManager.isLoginFlow.collectAsState(false).value
+
+    NavHost(
+        navController = navHostController,
+        startDestination =
+            if (isLogin) {
+                AgendaGraph
+            } else {
+                LoginGraph
+            },
```

Comment:
Here, you have two competing ways to determine where users must go when launching the app. To simplify this code, let's use the following model for our state:

```
data class MainState(
    val isLoggedIn: Boolean = false,
    val isCheckingAuth: Boolean = false,
)
```

And the ViewModel

```
class MainViewModel(
    private val sessionStorage: SessionStorage
) : ViewModel() {

    var state: MainState by mutableStateOf(MainState())
        private set

    init {
        viewModelScope.launch {
            state = state.copy(isCheckingAuth = true)
            state = state.copy(
                isLoggedIn = sessionStorage.getSession() != null
            )
            state = state.copy(isCheckingAuth = false)
        }
    }
}
```

Now we know if the user is logged in or not and when the user is being authenticated (checking to see if we have a session for them on the device). With these two pieces of data, we can update the UI to render based on the state.

```
val state = viewModel.state

if(!state.isCheckingAuth) {
  NavHost(
    navController = navController,
    startDestination = if (state.isLoggedIn) AgendaGraph else LoginGraph,
  ) {
    loginGraph(navController = navHostController)
    agendaGraph(navController = navHostController)
  }
}
```